executor/src/witgen/identity_processor.rs

```diff
@@ -42,21 +37,20 @@ impl<'a, 'c, T: FieldElement, Q: QueryCallback<T>> IdentityProcessor<'a, 'c, T,
     ) -> EvalResult<'a, T> {
         let result = match identity {
             Identity::Polynomial(identity) => self.process_polynomial_identity(identity, rows),
-            Identity::Lookup(LookupIdentity { left, id, .. })
-            | Identity::Permutation(PermutationIdentity { left, id, .. })
-            | Identity::PhantomLookup(PhantomLookupIdentity { left, id, .. })
-            | Identity::PhantomPermutation(PhantomPermutationIdentity { left, id, .. }) => {
-                self.process_lookup_or_permutation(*id, left, rows)
-            }
+            Identity::BusSend(bus_interaction) => self.process_lookup_or_permutation(
+                bus_interaction.id,
```

Comment:
I now see that this changes from an identity id to a bus/interaction id. Could you change the names of the parameters?

Reply:
Also shouldn't it be a T instead of a u64?

Reply:
oh wait, it's `bus_interaction.id` instead of `bus_interaction.interaction_id`.

Reply:
now `interaction_id` instead of `bus_id` ;)

Reply:
But yeah, in the future, we'll have to to change the machine call interface to receive a run-time bus ID instead of a static identity ID (which currently determines the bus ID uniquely and statically).